### PR TITLE
Drop Travis CI and completely migrate to Github Actions

### PR DESCRIPTION
**Pull request by itholic**
This PR proposes of drop Travis CI from now on.

IMO, I don't think there is a need to keep two management points for CI.

The following changes are includes in this PR.

- Removed the `.travis.yml`
- Added the existing tests in the `.travis.yml` to `master.yml` which is used for GitHub Actions.
- Removed badge for the Travis CI from `README.md`
- Renamed the filename from `download_travis_dependencies.sh` to `download_spark.sh` since this script is downloading the Spark only.
- Uncomment the test which was failing on a Travis CI.
- Renamed job names

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@ueshin WDYT?

**Comment by itholic** [comment]
@HyukjinKwon okay I'll check!

**Comment by itholic** [comment]
> Can you try to enable this test and see if passes?
> 
> ```
> databricks/koalas/tests/test_series.py:        # TODO: due to unknown reason, this test passes in Travis CI. Unable to reproduce in local.
> ```

Seems that It's working well 👍

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -12,13 +12,23 @@ jobs:
   pip_python35:
     name: PIP, Python (3.5)
```
Comment (HyukjinKwon) [inline .github/workflows/master.yml]: Can we name it as `PIP (Python, Spark, pandas, PyArrow)`?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -76,9 +86,17 @@ jobs:
         include:
```
Comment (HyukjinKwon) [inline .github/workflows/master.yml]: Can we also name it `Conda (Python, Spark, pandas, PyArrow)`?

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -12,13 +12,23 @@ jobs:
   pip_python35:
```
Comment (HyukjinKwon) [inline .github/workflows/master.yml]: Shall we also name it `pip_pytho35` -> `pip_build` or `pip`?

File: .github/workflows/master.yml (hunk 2 above)
Comment (HyukjinKwon) [inline .github/workflows/master.yml]: conda_python -> conda_build or conda

**Comment by itholic** [comment]
@HyukjinKwon Thanks! I just renamed them.

**Pull request merged**